### PR TITLE
docs/customizations: Move customization examples to dedicated folder

### PR DESCRIPTION
**Pull request by ArthurSens**
## Description

With the objective of improving our README, customization examples are being moved to a dedicated folder under `docs/`.

This is one of the changes being tracked at #1005 

## Type of change

_What type of changes does your code introduce to the kube-prometheus? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [X] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. Later this will be copied to the changelog file._

```release-note

```

### CODE REVIEW

**Comment by ArthurSens** [comment]
Do we want to add these to the website? I probably need to add the headers if that's true

File: docs/customizations/node-exporter-namespace.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+```jsonnet mdox-exec="cat examples/ksonnet-example.jsonnet"
+((import 'kube-prometheus/main.libsonnet') + {
+   values+:: {
+     nodeExporter+: {
+       namespace: 'my-custom-namespace',
+     },
+   },
+ }).nodeExporter.daemonset
+```
```
Comment (ArthurSens) [inline docs/customizations/node-exporter-namespace.md:13]: Some examples are pretty outdated or don't add much value, should we keep those?

File: docs/customizations/static-etcd-configuration.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+### Static etcd configuration
+
+In order to configure a static etcd cluster to scrape there is a simple [static-etcd.libsonnet](../../jsonnet/kube-prometheus/addons/static-etcd.libsonnet) mixin prepared - see [etcd.jsonnet](../../examples/etcd.jsonnet) for an example of how to use that mixin, and [Monitoring external etcd](../monitoring-external-etcd.md) for more information.
```
Comment (paulfantom) [inline docs/customizations/static-etcd-configuration.md:3]: Could you make it multiline? Also do we want to embed cited resources here for better readibility?

File: docs/customizations/node-exporter-namespace.md (hunk 1 above)
Comment (paulfantom) [reply docs/customizations/node-exporter-namespace.md:13]: let's remove such examples.

File: docs/customizations/internal-registry.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+$ docker push internal-registry.com/organization/alertmanager:v0.16.2
+...
+```
+
+The output of this command can be piped to a shell to be executed by appending `| sh`.
+
+Then to generate manifests with `internal-registry.com/organization`, use the `withImageRepository` mixin:
+
+```jsonnet mdox-exec="cat examples/internal-registry.jsonnet"
+local mixin = import 'kube-prometheus/addons/config-mixins.libsonnet';
```
Comment (paulfantom) [inline docs/customizations/internal-registry.md:23]: nit: we should probably change name of that file to something more descriptive.

File: examples/ksonnet-example.jsonnet [hunk 4]
```diff
@@ -1,9 +1,7 @@
...
-     daemonset+: {
-       metadata+: {
-         namespace: 'my-custom-namespace',
-       },
+   values+:: {
+     nodeExporter+: {
+       namespace: 'my-custom-namespace',
      },
    },
  }).nodeExporter.daemonset
```
Comment (paulfantom) [inline examples/ksonnet-example.jsonnet:7]: Can we remove this example and amend the doc about `Prometheus Object Name` to be more generic and include changing name and namespace for objects? The mechanism is the same for all objects not only Prometheus.

**Review by paulfantom — APPROVED** [review]
Looks good! Just one nit.

**Comment by ArthurSens** [comment]
What do you think of how it is looking now? :)

**Comment by paulfantom** [comment]
> Do we want to add these to the website? I probably need to add the headers if that's true

Yes, it would be good to add it to website. :+1: 

Side note: I am starting to wonder if we shouldn't split website into 2 and cross-link them :thinking: This way we could get rid of "website" repository which only synchronizes data from 2 other repos and just move sites into prometheus-operator repository and kube-prometheus. Probably something worth talking about in our meeting though.

**Review by paulfantom — APPROVED** [review]
lgtm

**Pull request merged**